### PR TITLE
Enable cbmc-concurrency regression tests by default

### PR DESCRIPTION
**Pull request by tautschnig**
For some reason we had not included these in our standard set of regression
tests, and sure enough they got out of sync and one actual regression happened.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b581af338d93ee942641f5ac45df1e3e6f9059a6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102230249

File: regression/cbmc-concurrency/Makefile [hunk 1]
```diff
@@ -1,10 +1,19 @@
 default: tests.log
 
+include ../../src/config.inc
+include ../../src/common
+
+ifeq ($(filter-out OSX MSVC,$(BUILD_ENV_)),)
+	# no POSIX threads on Windows
+	# for OSX we'd need sound handling of pointers in multi-threaded programs
```
Comment (smowton) [inline regression/cbmc-concurrency/Makefile:8]: ? Why is that concern particular to OSX?

**Review by smowton — APPROVED** [review]

File: regression/cbmc-concurrency/Makefile (hunk 1 above)
Comment (tautschnig) [reply regression/cbmc-concurrency/Makefile:8]: I believe the difference is whether `pthreadt` is a typedef for a pointer type or not, which varies from platform to platform.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 78437a18734021d3c08bd01efa4993c865d33e56).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102316841

**Pull request merged**